src/api/marking/marking.utils.ts

```diff
@@ -0,0 +1,162 @@
+
+// Constants
+
+// Included related data from lk and xref tables
+const markingIncludes = {
+  include: {
+    xref_taxon_marking_body_location: {
+      select: { body_location: true },
+    },
+    lk_marking_type: {
+      select: { name: true },
+    },
+    lk_marking_material: {
+      select: { material: true },
+    },
+    lk_colour_marking_primary_colour_idTolk_colour: {
+      select: { colour: true },
+    },
+    lk_colour_marking_secondary_colour_idTolk_colour: {
+      select: { colour: true },
+    },
+    lk_colour_marking_text_colour_idTolk_colour: {
+      select: { colour: true },
+    },
+  } satisfies Prisma.markingInclude,
+};
+
+// Schemas
+
+// Base schema for all marking related data
+const markingSchema = implement<marking>().with({
+  marking_id: zodID,
+  critter_id: zodID,
+  capture_id: zodID.nullable(),
+  mortality_id: zodID.nullable(),
+  taxon_marking_body_location_id: zodID,
+  marking_type_id: zodID.nullable(),
+  marking_material_id: zodID.nullable(),
+  primary_colour_id: zodID.nullable(),
+  secondary_colour_id: zodID.nullable(),
+  text_colour_id: zodID.nullable(),
+  identifier: z.string().nullable(),
+  frequency: z.number().nullable(),
+  frequency_unit: z.nativeEnum(frequency_unit).nullable(),
+  order: z.number().int().nullable(),
+  comment: z.string().nullable(),
+  attached_timestamp: z.coerce.date(),
+  removed_timestamp: z.coerce.date().nullable(),
+  ...zodAudit,
+});
+
+// Extended schema which has both base schema and included fields
+const markingIncludesSchema = implement<MarkingIncludes>().with({
+  ...markingSchema.shape,
+  lk_colour_marking_primary_colour_idTolk_colour: LookUpColourSchema.pick({
+    colour: true,
+  }).nullable(),
+  lk_colour_marking_secondary_colour_idTolk_colour: LookUpColourSchema.pick({
+    colour: true,
+  }).nullable(),
+  lk_colour_marking_text_colour_idTolk_colour: LookUpColourSchema.pick({
+    colour: true,
+  }).nullable(),
+  lk_marking_type: LookUpMarkingTypeSchema.pick({
+    name: true,
+  }).nullable(),
+  lk_marking_material: LookUpMaterialSchema.pick({
+    material: true,
+  }).nullable(),
+  xref_taxon_marking_body_location: XrefTaxonMarkingBodyLocationSchema.pick({
+    body_location: true,
+  }),
+});
+
+// Formatted API reponse schema which omits fields and unpacks nested data
+const markingResponseSchema = ResponseSchema.transform((obj) => {
+  const {
+    //omit
+    primary_colour_id,
+    secondary_colour_id,
+    text_colour_id,
+    marking_type_id,
+    marking_material_id,
+    taxon_marking_body_location_id,
+    //include
+    xref_taxon_marking_body_location,
+    lk_marking_type,
+    lk_marking_material,
+    lk_colour_marking_primary_colour_idTolk_colour,
+    lk_colour_marking_secondary_colour_idTolk_colour,
+    lk_colour_marking_text_colour_idTolk_colour,
+    ...rest
+  } = obj as MarkingIncludes;
+  return {
+    body_location: xref_taxon_marking_body_location?.body_location ?? null,
```

Comment:
I would throw ...rest before the additional properties. Doesnt really matter, but nice to have at least the primary key as the first value.